### ISSUE TITLE
QueryBuilderExtender ignores different alias for joined table

### ISSUE THREAD

**Issue by pk16011990** [shopsys/shopsys#1417]
### What is happening
When I call `QueryBuilderExtender::addOrExtendJoin()` with join which exists in query builder but with another alias, QueryBuilderExtender extends join instead of adding new join with different alias. Condition for new join contains new alias but It does not exist...

### Expected result
Passing test:
```php
public function testDifferentAlias()
{
    /** @var \Doctrine\ORM\EntityManager $entityManager */
    $entityManager = $this->getMockBuilder(EntityManagerInterface::class)
        ->disableOriginalConstructor()
        ->getMockForAbstractClass();

    $queryBuilder = new QueryBuilder($entityManager);
    $queryBuilder
        ->select('c')
        ->from(Category::class, 'c')
        ->join(Product::class, 'p', Join::WITH, 'p.id = 1');

    $queryBuilderExtender = new QueryBuilderExtender();
    $queryBuilderExtender->addOrExtendJoin($queryBuilder, Product::class, 'p2', 'p2.id = 2');

    $dql = $queryBuilder->getDQL();
    $this->assertSame('SELECT c FROM ' . Category::class . ' c INNER JOIN ' . Product::class . ' p WITH p.id = 1 INNER JOIN ' . Product::class . ' p2 WITH p2.id = 2', $dql);
}
```
not: 
```
1) Tests\FrameworkBundle\Unit\Component\Doctrine\QueryBuilderExtenderTest::testDifferentAlias
Failed asserting that two strings are identical.
--- Expected
+++ Actual
@@ @@
-'SELECT c FROM Shopsys\FrameworkBundle\Model\Category\Category c INNER JOIN Shopsys\ShopBundle\Model\Product\Product p WITH p.id = 1 INNER JOIN Shopsys\ShopBundle\Model\Product\Product p2 WITH p2.id = 2'
+'SELECT c FROM Shopsys\FrameworkBundle\Model\Category\Category c INNER JOIN Shopsys\ShopBundle\Model\Product\Product p WITH p.id = 1 WHERE p2.id = 2'

/var/www/html/packages/framework/tests/Unit/Component/Doctrine/QueryBuilderExtenderTest.php:36
```

**Comment by pk16011990** [comment]
remove stale label

**Comment by pk16011990** [comment]
remove stale

**Comment by s3tezsky** [comment]
fixed in #1971
